File: docsapp/com.ibm.concord.document.common/src/com/ibm/concord/document/common/pres/TaskContainer.java
package com.ibm.concord.document.common.pres;

import com.ibm.json.java.JSONObject;

public class TaskContainer extends Model
{

  private String content;

  public String getHTML()
  {
    String div = this.gartherAttrs(null);
    div += this.content;
    div += "</div>";
    return div;
  }

  public static TaskContainer fromJson(JSONObject obj)
  {
    TaskContainer tc = new TaskContainer();

    tc.id = obj.get("id").toString();

    if (obj.containsKey("content"))
      tc.content = obj.get("content").toString();

    if (obj.containsKey("attrs"))
    {
      tc.attrs = (JSONObject) obj.get("attrs");
    }
    return tc;
  }

  public String getContent()
  {
    return content;
  }

  public void setContent(String content)
  {
    this.content = content;
  }

}

File: docs-web.resources/com.ibm.docs.web.resources/WebContent/js/presentation/js/model/TaskContainer.js
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* U.S. Government Users Restricted Rights: Use, duplication or      */
/* disclosure restricted by GSA ADP Schedule Contract with IBM Corp. */
/*                                                                   */
/* ***************************************************************** */

dojo.provide("pres.model.TaskContainer");
dojo.require("pres.model.Htmlable");
dojo.require("pres.model.Attrable");
dojo.declare("pres.model.TaskContainer", [pres.model.Attrable, pres.model.Htmlable], {

	id: "",
	content: "",
	parent: null,

	constructor: function(taskContainer)
	{
		this.attrs = {};
		if(taskContainer)
		{
			this.id = taskContainer.id;
			this.attrs = taskContainer.attrs;
			this.content = taskContainer.content;
		}
	},

	toJson: function()
	{
		return {
			id: this.id,
			type: "taskContainer",
			content: this.content,
			attrs: dojo.clone(this.attrs)
		};
	},

	clone: function()
	{
		var taskContainer = new pres.model.TaskContainer();
		taskContainer.id = this.id;
		taskContainer.attrs = dojo.clone(this.attrs);
		taskContainer.content = this.content;
		return taskContainer;
	},

	getHTML: function()
	{
		var div = this._gartherAttrs();
		div += this.content;
		div += "</div>";
		return div;
	}

});
